torch/_prims/utils.py

```diff
@@ -1115,6 +1258,35 @@ def make_contiguous_strides_for(shape: ShapeType) -> Tuple[int, ...]:
     return result
 
 
+def make_channels_last_2d_strides_for(shape: ShapeType) -> Tuple[int, ...]:
+    # TODO: maybe inform the user of channels_last_3d if rank of the tensor is 5?
+    check(
+        len(shape) == 4,
+        lambda: "Only tensors of rank 4 can use the channels_last memory format",
+    )
+
+    shape = list(shape)
+    shape[1], shape[-1] = shape[-1], shape[1]
+    strides = list(make_contiguous_strides_for(shape))
```

Comment:
there's this weird inconsistency in eager for 0-element tensors where `contiguous` leaves the strides of following size-0 dimensions to be whatever the stride of the previous dimension was, whereas for channels-last that stride (and all subsequent) are zeroed:
```
In [41]: a=torch.empty(2,2,2,0, memory_format=torch.channels_last)

In [42]: a.stride()
Out[42]: (0, 1, 0, 2)

In [43]: a=torch.empty(2,2,2,0)

In [44]: a.stride()
Out[44]: (4, 2, 1, 1)
```
so using this approach would produce results that are different from eager. 
Another eager inconsistency is that empty tensors are always contiguous, but have to follow precise striding requirements to be channels_last contiguous, and we should probably fix that.

Reply:
Great catch -- I redid the implementation to capture this caveat and I'll add some additional testing